scripts/sam_uncollapse.pl

```diff
@@ -128,15 +128,16 @@ sub sam_uncollapse {
                         # Get QNAME
                         my ($id, $rest) = split /\t/, $line, 2;
                         # Find and remove appended copy number n
-                        $id =~ s/-(\d+)\Z//;
+                        $id =~ /^([^_-]+)-(\d+).?\d+/;
                         # Write appended copy number n to variable
-                        my $repeat = $1;
+                        my $read = $1;
+                        my $repeat = $2;
 			# If --suffix option is set...
 			if ($suffix) {
 				# Iterate over number of identical reads/alignments
 				for my $suffix (1..$repeat) {
 		                        # Recreate line with suffix
-                		        $line = join "\t", "$id.$suffix", $rest;
+                		        $line = join "\t", "$read.$suffix", $rest;
```

Comment:
At a quick glance, this actually looks quite different from the previous behavior. So different, in fact, that it might be a breaking change, i.e., that old calls of the script fed old inputs (with the expectations we had on the inputs back then) would result in different outputs. If that is indeed the case, I think it would be better to find a solution that is _not_ breaking, i.e., the script should behave in exactly the same way if faced with old inputs, but _also_ should be able to process the new input ID format according to our needs. If necessary, you could add a command-line option. But perhaps you can also manage without (and instead determine the input format and associated expected behavior automatically).

It would also be good if the PR description contained some concrete examples that highlight how the previous behavior (on old inputs) is unchanged, and how the change affects only new inputs.